### PR TITLE
Maintenance

### LINKED ISSUE #65: Update README
Readme needs a refresh as several things have changed since the last update.

### LINKED ISSUE #66: Bump FW versions
Some FWs (e.g. Qiskit) still have fairly old versions. Upgrade to the latest ones!

### PR DESCRIPTION
**Pull request by stroblme**
This PR bumps all fws to the latest version (including missing packages in `requirements.in`) (resolves #66) and contains some rework of the `README.md` (resolves #65 )

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,102 +40,107 @@ pre-commit install
...
 
 In summary, the following pipelines exist:
 - "prepare" : generates all possible combinations of configurations based on the current parameter set
-- "measure" : performs the actual time measurement by executing experiments for each of the previously generated configuration
-- "ctmeasure" : continous a previous time measurement
-- "visualize" : gathers all the experiment results and generates some nice plots
+- `measure` : performs the actual time measurement by executing experiments for each of the previously generated configuration with the ability to parallelize processing
+- `ctmeasure` : continous a previous time measurement
+- `combine` : gathers all the results from the `measure` pipeline and combines them into a single output dataset
+- `visualize` : takes the combined experiment results and generates some nice plots
```
Comment (eileen-kuehn) [inline README.md:65]: Here, the style is different from your first pipeline. The first one is quoted while these are marked as monospace. So please decide on a consistent format, e.g.

```suggestion
- "measure" : performs the actual time measurement by executing experiments for each of the previously generated configuration with the ability to parallelize processing
- "ctmeasure" : continous a previous time measurement
- "combine" : gathers all the results from the `measure` pipeline and combines them into a single output dataset
- "visualize" : takes the combined experiment results and generates some nice plots
```

File: README.md [hunk 2]
```diff
@@ -41,102 +40,107 @@ pre-commit install
...
 
 Note that is required to always run the "prepare" pipeline in advance to any actual processing pipeline.
 This is because of the current implementation relies on dynamically created nodes that are depending on the configuration and therefore requiring two separate pipeline executions.
 
 In summary, the following pipelines exist:
 - "prepare" : generates all possible combinations of configurations based on the current parameter set
-- "measure" : performs the actual time measurement by executing experiments for each of the previously generated configuration
-- "ctmeasure" : continous a previous time measurement
-- "visualize" : gathers all the experiment results and generates some nice plots
+- `measure` : performs the actual time measurement by executing experiments for each of the previously generated configuration with the ability to parallelize processing
```
Comment (eileen-kuehn) [inline README.md:62]: Grammar mistake:

```suggestion
- `measure` : performs the actual time measurement by executing experiments for each of the previously generated configurations with the ability to parallelize processing
```

File: README.md [hunk 3]
```diff
@@ -41,102 +40,107 @@ pre-commit install
...
+- `ctmeasure` : continous a previous time measurement
```
Comment (eileen-kuehn) [inline README.md:63]: I don't get this one. What do you mean here?

File: README.md (hunk 1 above)
Comment (eileen-kuehn) [inline README.md:65]: Implicit rating :)

```suggestion
- `visualize` : takes the combined experiment results and generates your plots
```

File: README.md [hunk 4]
```diff
@@ -41,102 +40,107 @@ pre-commit install
...
+To use this, you have to explicitly call the individual pipelines.
+In summary this will look as follows:
 ```
-poetry run kedro run --pipeline "measure" --env dask --runner quafel.runner.DaskRunner
+kedro run --pipeline prepare
+kedro run --pipeline measure --env dask --runner quafel.runner.MyParallelRunner
+kedro run --pipeline combine
+kedro run --pipeline visualize
 ```
 which will calculate the duration and result for each configuration.
```
Comment (eileen-kuehn) [inline README.md:79]: This does not make it properly clear. Please consider explicitly mentioning, that the "measure" pipeline is executed in a multiprocessing style. You might also consider adding a note, why this is not possible for one of the other pipelines.

File: README.md [hunk 5]
```diff
@@ -41,102 +40,107 @@ pre-commit install
...
-See [Dask Setup](#runner-dask-setup) for detail on this.
-For details on the output, see the [Data Structure Section](#floppy_disk-data-structure).
 
+For details on the output, see the [Data Structure Section](#floppy_disk-data-structure).
 
-***
-:construction: only:
+Note that if you want to re-run e.g. the `visualize` pipeline, you have to re-run the `prepare` pipeline as well!.
```
Comment (eileen-kuehn) [inline README.md:83]: Please consider adding a reason for this as this does not seem to be intuitive.

File: README.md [hunk 6]
```diff
@@ -41,102 +40,107 @@ pre-commit install
...
-  - results are zero padded, so it is ensured that also state combinations with $0$ probability are represented.
+  - Simulator results of the job with the corresponding id.
+  - Result formats are unified as a dictionary with the keys containing the binary bit representation of the measured qubit and the normalized counts as values.
+  - Results are zero padded, so it is ensured that also state combinations with $0$ probability are represented.
 - [data/05_execution_durations](data/05_execution_durations/):
-  - duration for the simulation of the job with the corresponding id.
-  - duration is only measured for the execution of the simulator
-  - combining and post-processing results (to obtain the dictionary representation) is not involved
+  - Duration for the simulation of the job with the corresponding id.
+  - Duration is only measured for the execution of the simulator
```
Comment (eileen-kuehn) [inline README.md:131]: What do you mean here? What is skipped from measuring the duration and why do you need two points on this?

File: README.md [hunk 7]
```diff
@@ -41,102 +40,107 @@ pre-commit install
...
+  - Combining and post-processing results (to obtain the dictionary representation) is not involved
```
Comment (eileen-kuehn) [inline README.md:132]: What do you mean with involved? Isn't it about what is *contained* in those folders?

File: README.md [hunk 8]
```diff
@@ -162,15 +166,15 @@ This dictionary is required to contain all combinations of bitstrings that resul
...
-poetry run kedro run --pipeline "measure" --env dask --runner quafel.runner.DaskRunner
+poetry run kedro run --pipeline `measure` --env dask --runner quafel.runner.DaskRunner
 ```
 without any additional configuration, Kedro creates Dask scheduler and also four worker nodes.
 This behavior can be controlled in [conf/dask/parameters.yml](conf/dask/parameters.yml).
 Setting the address parameter will cause Kedro trying to connect to an existing scheduler at the specified address.
 You create scheduler and $N$ workers by running `.vscode/spawn_n_workers.sh -N` from the root folder of the project.
-Alternatively set `n_workers` in [conf/dask/parameters.yml](conf/dask/parameters.yml) and comment out `address` to specify the number of workers that kedro should spawn.
\ No newline at end of file
+Alternatively set `n_workers` in [conf/dask/parameters.yml](conf/dask/parameters.yml) and comment out `address` to specify the number of workers that kedro should spawn. -->
```
Comment (eileen-kuehn) [inline README.md:180]: Please remove in case you don't want it anymore.

**Review by eileen-kuehn — CHANGES_REQUESTED** [review]
Nice work 🥳 , please look at the comments I left before things can get merged.

File: README.md (hunk 5 above)
Comment (stroblme) [reply README.md:86]: explained and created issue #69

File: README.md [hunk 9]
```diff
@@ -41,102 +40,111 @@ pre-commit install
...
 This is because of the current implementation relies on dynamically created nodes that are depending on the configuration and therefore requiring two separate pipeline executions.
 
 In summary, the following pipelines exist:
-- "prepare" : generates all possible combinations of configurations based on the current parameter set
-- "measure" : performs the actual time measurement by executing experiments for each of the previously generated configuration
-- "ctmeasure" : continous a previous time measurement
-- "visualize" : gathers all the experiment results and generates some nice plots
+- `prepare` : generates all possible combinations of configurations based on the current parameter set
+- `measure` : performs the actual time measurement by executing experiments for each of the previously generated configurations with the ability to parallelize processing
+- `ctmeasure` : if the `measure` pipeline failed of was cancelled, use this pipeline to resume
```
Comment (eileen-kuehn) [inline README.md:63]: Small typo:

```suggestion
- `ctmeasure` : if the `measure` pipeline failed or was cancelled, use this pipeline to resume
```

File: README.md [hunk 10]
```diff
@@ -41,102 +40,111 @@ pre-commit install
...
 
+Here, only the pipeline `measure` will utilize multiprocessing and the rest will run single process.
+We recommend this approach since there is no advantage by running the other pipelines in parallel as well.
+Of course, you can run the `measure` pipeline in a single process as well by omitting the `--runner` option.
+
+For details on the output, see the [Data Structure Section](#floppy_disk-data-structure).
 
-***
-:construction: only:
+Note that if you want to re-run e.g. the `visualize` pipeline, you have to re-run the `prepare` pipeline as well!.
```
Comment (eileen-kuehn) [inline README.md:86]: Please decide if you want to use the period or exclamation mark :)

File: README.md [hunk 11]
```diff
@@ -41,102 +40,111 @@ pre-commit install
...
-
-### :floppy_disk: Data Structure
+## Data Structure :floppy_disk:
 
 - [data/01_raw](data/01_raw):
-  - [Evaluation Matrix](data/01_raw/dataset.json) containing all valid values for ```frameworks```, ```qubits```, ```depths```, and ```shots``` as specified in the [data_generation.yml](conf/base/parameters/data_generation.yml) file.
+  - **Versioned** [Evaluation Matrix](data/01_raw/dataset.json) containing all valid values for ```frameworks```, ```qubits```, ```depths```, and ```shots``` as specified in the [data_generation.yml](conf/base/parameters/data_generation.yml) file.
 - [data/02_intermediate](data/02_intermediate):
-  - Evaluation Partitions split into single ```.csv``` files
+  - Evaluation Partitions split into single ```.csv``` files.
```
Comment (eileen-kuehn) [inline README.md:126]:
```suggestion
  - Evaluation partitions split into single ```.csv``` files.
```

**Review by eileen-kuehn — CHANGES_REQUESTED** [review]
Nearly done, just three very small changes and then you are good to go 👍

**Pull request merged**